pyramda/iterable/flatten.py

```diff
@@ -0,0 +1,26 @@
+from collections import Iterable
+from past.builtins import basestring
+
+from pyramda.function.curry import curry
+
+
+@curry
+def flatten(xs, depth):
```

Comment:
Yes, I think it would be simpler to not enforce a recursion depth. If a caller wants to only flatten up to a certain depth, that can still be done by first transforming the tree and adding depth markers to each node. That's probably not worth supporting in Pyramda directly though.